### ISSUE TITLE
Warm-start skips: ~12–14% of reproductions fail with incompatible_state (policy shape mismatch)

### ISSUE THREAD

**Issue by csmangum** [Dooders/AgentFarm#961]
## Context

Observed during the learning-positive inheritance-ladder pilot (`experiments/inheritance_ab_learning_positive`, balanced profile, 3000 steps, pop 8 → cap 32) tracked in #904.

## Observation

Across P1–P3 arms, **~12–14% of warm-start attempts are skipped** with reason `incompatible_state`:

| Arm | seed 42 | seed 7 |
|-----|---------|--------|
| lamarckian | 15/110 skipped (86%) | 13/104 skipped (88%) |
| p2 | 19/133 skipped (86%) | 14/112 skipped (88%) |
| p3 | 22/161 skipped (86%) | 14/113 skipped (88%) |

All skips use `incompatible_state`; no other skip reasons observed.

## Mechanism

In `farm/core/policy_inheritance.py`, `_policy_state_is_compatible()` rejects transfer when parent/child policy `state_dict` keys or tensor shapes differ. This happens when chromosome mutation changes network architecture (e.g. `dqn_hidden_size`) between parent and offspring.

Affected offspring **fall back to cold start**, diluting the treatment arm and making P1–P4 comparisons noisier than telemetry `warmstart_coverage` alone suggests.

## Impact

- Treatment effect is diluted: ~1 in 7 reproductions does not receive the intended payload.
- Reported `warmstart_coverage` (~86–88%) overstates effective inheritance when architecture drift is frequent.

## Possible directions

- Gate warm-start on architecture compatibility (only attempt when shapes match; already happens, but could surface in offspring telemetry).
- Per-offspring warm-start status in DB/metadata so early-life analysis can stratify warm-started vs cold-fallback offspring.
- Reduce architecture mutation rate during inheritance A/B, or hold network topology fixed for the ladder experiment.
- Document expected skip rate under current mutation settings.

## Reproduce

```bash
PYTHONHASHSEED=0 python scripts/run_inheritance_mode_ab.py \
  --arms lamarckian p2 p3 \
  --population 8 --max-population 32 \
  --profiles balanced --seeds 42 7 \
  --num-steps 3000 --disk-database
```

Check `policy_inheritance_metrics.warmstart_skipped_reasons` in each run's `intrinsic_evolution_metadata.json`.
